TowerNinja/Assets/Scripts/enemyDmgHitbox.cs

```diff
@@ -114,12 +117,31 @@ public void DamageEnemy(int damage)
         if (_healthPoint <= MinHealthPoint)
         {
             EnemyWave.spawnedEnemy--;
+
+            // give mana reward
+            GiveManaReward();
+
             //Debug.Log(this.gameObject + "is destroyed.");
             Destroy(this.gameObject);
             ReportEnemyDeath();
         }
     }
 
+    private void GiveManaReward()
+    {
+        GameObject resourceManagerObject = GameObject.Find("ResourceManager").gameObject;
+        ResourceManager resourceManager = resourceManagerObject.GetComponent<ResourceManager>();
+
+        if (gameObject.name.StartsWith("Big"))
+        {
+            resourceManager.IncreaseMana(BigEnemyManaReward);
+        }
+        else
+        {
+            resourceManager.IncreaseMana(EnemyManaReward);
+        }
```

Comment:
It's prolly best to set the EnemyManaReward from the unity via prefab.
Then we can just do 1 line